### PR TITLE
langid+wtl

### PR DESCRIPTION
**Pull request by a-rios**

### CODE REVIEW

File: compare_mt/compare_mt_main.py [hunk 1]
```diff
@@ -429,6 +470,12 @@ def main():
                       help="Number of decimals to print for floating point numbers")
   parser.add_argument('--scorer_scale', type=float, default=100, choices=[1, 100],
                       help="Set the scale of BLEU, METEOR, WER and chrF to 0-1 or 0-100 (default 0-100)")
+  parser.add_argument('--lang_id', type=str, nargs='*',
+                      default=['model=model,min_length=min_length,print_lines=False,print_line_numbers=False'],
```
Comment (bricksdont) [inline compare_mt/compare_mt_main.py]: Do I understand correctly that the default is to include the lang id report? How do I turn it off?

File: compare_mt/compare_mt_main.py [hunk 2]
```diff
@@ -364,6 +367,44 @@ def generate_sentence_examples(ref, outs, src=None,
   reporter.generate_report()
   return reporter 
 
+def generate_lang_id_report(ref, outs, model,
+                            min_length=5,
+                            print_lines=False,
+                            print_line_numbers=False):
+    wtl = WhatTheLang()
```
Comment (bricksdont) [inline compare_mt/compare_mt_main.py]: You could make this conditional; only create the object when wtl is the lang id method.

File: compare_mt/compare_mt_main.py [hunk 3]
```diff
@@ -364,6 +367,44 @@ def generate_sentence_examples(ref, outs, src=None,
...
+    for out in outs:
+        langs = defaultdict(int)
+        lang_lines = defaultdict(list)
+        lang_line_numbers = defaultdict(list)
+        for i, sentence in enumerate(out, start=1):
+            line = corpus_utils.list2str(sentence)
+            if len(sentence) >= int(min_length):
+                if model=="langid":
+                    (lang, prob) = langid.classify(line)
+                elif model=="wtl":
```
Comment (bricksdont) [inline compare_mt/compare_mt_main.py:389]: If `model` is neither "wtl" nor "langid", this should be caught somewhere. For instance, with a NotImplementedError:

```python
if model=="langid":
    (lang, prob) = langid.classify(line)
elif model=="wtl":
    lang = wtl.predict_lang(line)
else:
    raise NotImplementedError(f"Unknown model for language identification: '{model}'.")
```

**Review by bricksdont — CHANGES_REQUESTED** [review]

**Comment by bricksdont** [comment]
Otherwise looks good! Could you please also increment the minor version in https://github.com/ZurichNLP/compare-mt/blob/master/compare_mt/__init__.py#L12?

File: compare_mt/compare_mt_main.py (hunk 1 above)
Comment (a-rios) [reply compare_mt/compare_mt_main.py]: done

File: compare_mt/compare_mt_main.py (hunk 2 above)
Comment (a-rios) [reply compare_mt/compare_mt_main.py]: done

File: compare_mt/compare_mt_main.py (hunk 3 above)
Comment (a-rios) [reply compare_mt/compare_mt_main.py:389]: done

**Review by bricksdont — APPROVED** [review]

**Pull request merged**